datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -194,6 +203,71 @@ impl FileScanConfig {
             .with_repartition_file_min_size(repartition_file_min_size)
             .repartition_file_groups(&file_groups)
     }
+
+    /// Attempts to do a bin-packing on files into file groups, such that any two files
+    /// in a file group are ordered and non-overlapping with respect to their statistics.
+    /// It will produce the smallest number of file groups possible.
+    pub fn sort_file_groups(
```

Comment:
maybe we could name this function something more related to the non overlapping nature

Perhaps like `split_groups_by_statistics` or something?

Reply:
I think that is a good name, I'll go with that